Receptai.client/components/Banner/Banner.vue

```diff
@@ -2,46 +2,140 @@
 import axios from "axios";
 import RecipeBannerComponent from "@/components/Banner/RecipeBannerComponent.vue";
 
-const config = useRuntimeConfig();
-
 interface Recipe {
     id: number;
     name: string;
     shortDescription: string;
+    author: Author;
+    dateCreated: string;
+    dateModified: string | null;
     previewImage: string;
+    tutorialVideo?: string;
+    ingredients: Ingredients[];
+    instructions: string[];
+    tags: string[];
+    categories: Category[];
+    minutesToPrepare: number;
+    portions: number;
+}
+
+interface Author {
+    name: string;
+}
+
+interface Ingredients {
+    purpose: string;
+    ingredients: Ingredient[];
+}
+
+interface Ingredient {
+    name: string;
+    quantity: number;
+    unit: string;
+}
+
+interface Category {
+    id: number;
+    name: string;
+    description: string | null;
+    previewImageUrl: string | null;
+}
+
+const recipeList = ref<Recipe[]>([]);
+const bannerOffset = ref(0);
+const bannerComponentWidth = 192;
```

Comment:
I don't fully understand this approach, full window width can accessed with javascript: window.innerWidth. So, maybe this can be done dynamically?